src/table/utils/handle-key-press.ts

```diff
@@ -54,35 +54,42 @@ export const getNextCellCoord = (
   evt: React.KeyboardEvent,
   rootElement: HTMLElement,
   cellCoord: [number, number],
-  topAllowedRow = 0
+  allowedRows: {
+    selected: {
+      top: number;
+      bottom: number;
+    };
+    top: number;
+    bottom: number;
+  } = { selected: { top: 0, bottom: 0 }, top: 2, bottom: 1 }
 ): [number, number] => {
   const rowCount = rootElement.getElementsByClassName('sn-table-row').length;
   const columnCount = rootElement.getElementsByClassName('sn-table-head-cell').length;
   let [nextRow, nextCol] = cellCoord;
 
   switch (evt.key) {
     case 'ArrowDown':
-      nextRow < rowCount - 1 && nextRow++;
+      nextRow < rowCount - 1 - allowedRows.selected.bottom && nextRow++;
       break;
     case 'ArrowUp':
-      nextRow > topAllowedRow && nextRow--;
+      nextRow > allowedRows.selected.top && nextRow--;
       break;
     case 'ArrowRight':
-      // topAllowedRow greater than 0 means we are in selection mode
-      if (topAllowedRow > 0) break;
+      // allowedRows.top greater than 0 means we are in selection mode
+      if (allowedRows.selected.top > 0) break;
       if (nextCol < columnCount - 1) {
         nextCol++;
-      } else if (nextRow < rowCount - 1) {
+      } else if (nextRow < rowCount - 1 - allowedRows.bottom) {
         nextRow++;
         nextCol = 0;
       }
       break;
     case 'ArrowLeft':
-      // topAllowedRow greater than 0 means we are in selection mode
-      if (topAllowedRow > 0) break;
+      // allowedRows.top greater than 0 means we are in selection mode
+      if (allowedRows.selected.top > 0) break;
       if (nextCol > 0) {
         nextCol--;
-      } else if (nextRow > 0) {
+      } else if (nextRow > allowedRows.top) {
```

Comment:
same here, you left arrow is blocked in selection mode

Reply:
this line should still be reverted

Reply:
Sorry for these mistakes, things went mess when I reverted the last changes, I see my selection mode testing scenario is gone and this guy is left here. Thanks for considering them. I'll fix it.